src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1476,29 +1476,133 @@ def _configure_security_policy_storage_params(arg_ctx):
+            'This can be same managed instance, or another managed instance on same subscription. '
+            'When not specified it defaults to source managed instance.')
+
+        c.argument(
+            'target_resource_group_name',
+            options_list=['--dest-resource-group'],
+            help='Name of the resource group of the managed instance to restore managed database to. '
+            'When not specified it defaults to source resource group.')
+
     with self.argument_context('sql midb short-term-retention-policy set') as c:
         create_args_for_complex_type(
             c, 'parameters', ManagedDatabase, [
                 'deleted_time',
                 'retention_days'
             ])
 
-        c.argument('deleted_time',
-                   options_list=['--deleted-time'],
-                   help='If specified, updates retention days for a deleted database, instead of an existing database.'
-                   'Must match the deleted time of a deleted database on the source Managed Instance.')
+        c.argument(
+            'deleted_time',
+            options_list=['--deleted-time'],
+            help='If specified, updates retention days for a deleted database, instead of an existing database.'
+            'Must match the deleted time of a deleted database on the source Managed Instance.')
 
-        c.argument('retention_days',
-                   options_list=['--retention-days'],
-                   required=True,
-                   help='New backup short term retention policy in days.'
-                   'Valid policy for live database is 7-35 days, valid policy for dropped databases is 0-35 days.')
+        c.argument(
+            'retention_days',
+            options_list=['--retention-days'],
+            required=True,
+            help='New backup short term retention policy in days.'
+            'Valid policy for live database is 7-35 days, valid policy for dropped databases is 0-35 days.')
 
     with self.argument_context('sql midb short-term-retention-policy show') as c:
-        c.argument('deleted_time',
-                   options_list=['--deleted-time'],
-                   help='If specified, shows retention days for a deleted database, instead of an existing database.'
-                   'Must match the deleted time of a deleted database on the source Managed Instance.')
+        c.argument(
+            'deleted_time',
+            options_list=['--deleted-time'],
+            help='If specified, shows retention days for a deleted database, instead of an existing database.'
+            'Must match the deleted time of a deleted database on the source Managed Instance.')
+
+    with self.argument_context('sql midb long-term-retention-policy set') as c:
+        create_args_for_complex_type(
+            c, 'parameters', ManagedDatabase, [
+                'weekly_retention',
+                'monthly_retention',
+                'yearly_retention',
+                'week_of_year'
```

Comment:
can we just use one parameter to define retention days with unit defined?

Reply:
It's a little more complex than that.  "Weekly retention" means how long each weekly backup should be kept; "monthly retention" means how long each monthly backup should be kept; "yearly backup" means how long to keep a backup that happens during "week_of_year". 

More info: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-long-term-retention or just message me if this is confusing!

Reply:
With the link you provided, it becomes clear to me now. So can we consider refine the help message for these parameters and provide the link for more information?

Reply:
Sure.  I can add those

Reply:
discussed on Teams